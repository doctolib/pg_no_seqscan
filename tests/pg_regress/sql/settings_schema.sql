-- Test schema filtering

SET pg_no_seqscan.level = ERROR;

CREATE SCHEMA test_schema1;
CREATE SCHEMA test_schema2;
CREATE TABLE test_schema1.foo AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE test_schema2.bar AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE public.baz AS (SELECT * FROM generate_series(1,10) as id);

-- Set check_schemas to only check test_schema1 and public
SET pg_no_seqscan.check_schemas = 'test_schema1,public';

-- This should be ignored due to schema not in check_schemas setting
SELECT * FROM test_schema2.bar;

-- These should error
SELECT * FROM test_schema1.foo;

-- Cleanup
DROP TABLE test_schema1.foo, test_schema2.bar, public.baz;
DROP SCHEMA test_schema1, test_schema2;
