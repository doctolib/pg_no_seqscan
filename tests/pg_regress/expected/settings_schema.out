-- Test schema filtering
SET pg_no_seqscan.level = ERROR;
CREATE SCHEMA test_schema1;
CREATE SCHEMA test_schema2;
CREATE TABLE test_schema1.foo AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE test_schema2.bar AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE public.baz AS (SELECT * FROM generate_series(1,10) as id);
-- Set check_schemas to only check test_schema1 and public
SET pg_no_seqscan.check_schemas = 'test_schema1,public';
-- This should be ignored due to schema not in check_schemas setting
SELECT * FROM test_schema2.bar;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

-- These should error
SELECT * FROM test_schema1.foo;
ERROR:  A 'Sequential Scan' on foo has been detected.
  - Run an EXPLAIN on your query to check the query plan.
  - Make sure the query is compatible with the existing indexes.

Query: SELECT * FROM test_schema1.foo;

-- Cleanup
DROP TABLE test_schema1.foo, test_schema2.bar, public.baz;
DROP SCHEMA test_schema1, test_schema2;
